src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -305,39 +356,64 @@ asf::auto_release_ptr<asr::Project> AppleseedInteractiveRender::prepare_project(
     return project;
 }
 
-void AppleseedInteractiveRender::update_camera_parameters(INode* camera)
+void AppleseedInteractiveRender::update_camera_object(INode* camera)
 {
     ViewParams view_params;
     get_view_params_from_view_node(view_params, camera, m_time);
 
-    asr::ParamArray& params = m_project->get_scene()->get_active_camera()->get_parameters();
-    params.insert("horizontal_fov", asf::rad_to_deg(view_params.fov));
-
-#if MAX_RELEASE >= 18000
-    MaxSDK::IPhysicalCamera* phys_camera = dynamic_cast<MaxSDK::IPhysicalCamera*>(camera->EvalWorldState(m_time).obj);
-
-    if (phys_camera && phys_camera->GetDOFEnabled(m_time, FOREVER))
-    {
-        set_camera_dof_params(params, phys_camera, m_bitmap, m_time);
-    }
-#endif
+    auto new_camera = build_camera(camera, view_params, m_bitmap, RendererSettings::defaults(), m_time);
+    get_render_session()->get_render_controller()->schedule_update(
+        std::unique_ptr<ScheduledAction>(new CameraObjectUpdateAction(m_project.ref(), new_camera)));
 }
 
-void AppleseedInteractiveRender::update_camera_transform(INode* camera)
+void AppleseedInteractiveRender::update_render_view()
 {
+    auto time = GetCOREInterface()->GetTime();
+
+    ViewExp& view_exp = GetCOREInterface()->GetActiveViewExp();
+
+    if (!view_exp.IsAlive())
+        return;
+
+    ViewExp13* vp13 = reinterpret_cast<ViewExp13*>(view_exp.Execute(ViewExp::kEXECUTE_GET_VIEWEXP_13));
+    INode* view_camera = vp13->GetViewCamera();
+
     ViewParams view_params;
-    get_view_params_from_view_node(view_params, camera, m_time);
+    if (view_camera != nullptr)
+        get_view_params_from_view_node(view_params, view_camera, m_time);
+    else
+        get_view_params_from_viewport(view_params, view_exp, m_time);
+
+    auto new_camera = build_camera(view_camera, view_params, m_bitmap, RendererSettings::defaults(), m_time);
+    get_render_session()->get_render_controller()->schedule_update(
+        std::unique_ptr<ScheduledAction>(new CameraObjectUpdateAction(m_project.ref(), new_camera)));
 
-    m_project->get_scene()->get_active_camera()->transform_sequence().set_transform(
-        static_cast<float>(m_time),
-        asf::Transformd::from_local_to_parent(to_matrix4d(Inverse(view_params.affineTM))));
+    if (view_camera != nullptr)
+        m_node_callback.reset(new SceneChangeCallback(this, view_camera));
 }
 
 InteractiveSession* AppleseedInteractiveRender::get_render_session()
 {
     return m_render_session.get();
 }
 
+AppleseedInteractiveRender* AppleseedInteractiveRender::m_current_interactive;
+
+VOID CALLBACK AppleseedInteractiveRender::timer_proc(
```

Comment:
I'm not sure why this method is defined here, to be honest. Why not define it in `ViewportCallback` which appears to be the only class using it?

Reply:
Moved.